language-testutils/src/test/java/de/jplag/testutils/datacollector/TokenPositionTestData.java

```diff
@@ -40,22 +44,46 @@ private void readFile(File testFile) throws IOException {
         int currentLine = 0;
 
         for (String sourceLine : testFileLines) {
-            if (sourceLine.charAt(0) == '>') {
-                this.sourceLines.add(sourceLine.substring(1));
-                currentLine++;
-            }
+            if (!sourceLine.isBlank()) {
+                switch (sourceLine.charAt(0)) {
+                    case '>' -> {
+                        this.sourceLines.add(sourceLine.substring(1));
+                        currentLine++;
+                    }
+
+                    case '$' -> {
+                        this.extractTokenData(sourceLine, currentLine);
+                        int column = sourceLine.indexOf('|');
+                        String[] tokenDescriptionParts = sourceLine.split(" ", 0);
+
+                        String typeName = tokenDescriptionParts[tokenDescriptionParts.length - 2];
+                        int length = Integer.parseInt(tokenDescriptionParts[tokenDescriptionParts.length - 1]);
+                        this.expectedTokens.add(new TokenData(typeName, currentLine, column, length));
+                    }
 
-            if (sourceLine.charAt(0) == '$') {
-                int column = sourceLine.indexOf('|');
-                String[] tokenDescriptionParts = sourceLine.split(" ", 0);
+                    case '#' -> {
+                        // Line is considered a comment
+                    }
 
-                String typeName = tokenDescriptionParts[tokenDescriptionParts.length - 2];
-                int length = Integer.parseInt(tokenDescriptionParts[tokenDescriptionParts.length - 1]);
-                this.expectedTokens.add(new TokenData(typeName, currentLine, column, length));
+                    default -> Assertions.fail("Invalid line for token position test " + this.descriptor + " at line: " + sourceLine);
+                }
             }
         }
     }
 
+    private void extractTokenData(String line, int currentSourceLine) {
+        if (!line.matches(TOKEN_DEFINITION_LINE_REGEX)) {
+            Assertions.fail("Invalid line for token position test " + this.descriptor + " at line: " + line);
+        }
+
+        int column = line.indexOf('|');
```

Comment:
We have quite a few chars with specific meaning in this PR (>, $, #, |). Can you introduce well-named constants for them?